x-pack/plugins/apm/public/locator/locator.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { IUiSettingsClient } from '@kbn/core/public';
+import { LocatorDefinition } from '@kbn/share-plugin/common';
+import { UI_SETTINGS } from '@kbn/data-plugin/public';
+import type { TimePickerTimeDefaults } from '../components/shared/date_picker/typings';
+import type { APMLocatorPayload } from './helpers';
+
+const helpersModule = import('./helpers');
+
+export const APM_APP_LOCATOR_ID = 'APM_LOCATOR';
+
+export class APMLocatorDefinition
```

Comment:
This only links to one(-ish) page. Should we make it clear in the locator name that that is the case? Given the arguments that this function expects (parameters specific to the service detail pages), I don't think we can sensibly extend it to cover other pages as well, so the name should probably reflect that.